test/gtest/ucp/test_ucp_proto.cc

```diff
@@ -57,6 +57,7 @@ ucp_md_map_t test_ucp_proto::get_md_map(ucs_memory_type_t mem_type)
         const uct_md_attr_t *md_attr = &context()->tl_mds[md_index].attr;
         if ((md_attr->cap.flags & UCT_MD_FLAG_REG) &&
             (md_attr->cap.reg_mem_types & UCS_BIT(mem_type)) &&
+            (md_attr->cap.cache_mem_types & UCS_BIT(mem_type)) &&
```

Comment:
why needed?

Reply:
it is used to test `ucp_datatype_iter_mem_reg()` which only work for cacheable MDs

Reply:
why is it so?
will it also be the case for proto v2 and when we replace dt_reg by ucp_memh for protov1?

Reply:
yes, that is design
`ucp_datatype_iter_mem_reg()` call `ucp_memh_get()` which get memh from rcache and we don't want _selectively exceluded MDs_ in rcache.

Reply:
`/* ucp_datatype_iter_mem_reg() always goes directly to registration cache */`